crawl/pom.py

```diff
@@ -564,6 +584,9 @@ def _gen_dependencies(self, pomcontenttype):
         # account for any version overrides that need to carry over to the
         # Maven build.
         transitives = self._get_transitive_deps(self.dependencies)
+
+        # Update the dependencies according to the overridden file
+        transitives = self._update_overridden_deps(transitives)
```

Comment:
Is this necessary, since the transitives are computed based on self.dependencies?

Reply:
Yes, the if the direct deps includes transitives which need to be overridd.en